UPGRADE.rst

```diff
@@ -85,6 +85,19 @@ for example:
      wget https://packages.matrix.org/debian/pool/main/m/matrix-synapse-py3/matrix-synapse-py3_1.3.0+stretch1_amd64.deb
      dpkg -i matrix-synapse-py3_1.3.0+stretch1_amd64.deb
 
+Upgrading to v1.34.0
+====================
+
+Deprecation of the built-in account validity feature
+----------------------------------------------------
+
+As of Synapse v1.34.0, the built-in account validity feature is deprecated in favour of
+plugin modules. If you are using this feature and want to keep its current behaviour,
+please use the `synapse-email-account-validity <https://github.com/matrix-org/synapse-email-account-validity>`_
+module.
```

Comment:
I think we'll need clearer instructions for admins here: how do they install the module, how should they update the config?

Reply:
These instructions are available at https://github.com/matrix-org/synapse-email-account-validity, my idea was that admins would follow the link to the module and follow the steps described there, but maybe that's not clear enough?

Reply:
wellll... I think on the one hand, we need some migration instructions rather than instructions that start from scratch.

Also, `pip install` doesn't work for people who didn't install with pip. (docker, debian, etc, etc)

Reply:
> I think on the one hand, we need some migration instructions rather than instructions that start from scratch.

The module attempts a migration when starting fwiw so just installing it from scratch and replacing the `account_validity` config with the module's should be enough, but fair enough I'll add more instructions on this doc as well.

> Also, `pip install` doesn't work for people who didn't install with pip. (docker, debian, etc, etc)

True, however I don't know how we support non-pip installs since none of the other modules we publish seem to do.

Reply:
I'm a bit uncertain what this PR should be doing to fix this issue. We had a discussion about it in #synapse-dev and (I think?) came to a conclusion (I don't remember the specifics, but iirc it was about telling users how to build custom Docker images based on the Synapse one with the modules they want in it) but it feels a bit out of scope for this PR to me.

Reply:
I appreciate it feels like scope creep, but I think it's a prerequisite for moving functionality out of core Synapse into modules, which is why I questioned whether that was a good decision.

In short: if we're going down that path, I'd like us to have solutions in place for modules with docker *and debian* before this PR lands.

Reply:
ftr we decided to ditch the deprecation issue for now, so this PR isn't blocked on finding a solution to this problem.